src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java

```diff
@@ -658,39 +658,47 @@ public MemberImpl createMember(GuildImpl guild, DataObject memberJson, DataObjec
 
         // Load voice state and presence if necessary
         if (voiceStateJson != null && member.getVoiceState() != null)
-            createVoiceState(guild, voiceStateJson, user, member);
+            createGuildVoiceState(member, voiceStateJson);
         if (presence != null)
             createPresence(member, presence);
         return member;
     }
 
-    private void createVoiceState(GuildImpl guild, DataObject voiceStateJson, User user, MemberImpl member)
-    {
+    public GuildVoiceState createGuildVoiceState(MemberImpl member, DataObject voiceStateJson) {
         GuildVoiceStateImpl voiceState = (GuildVoiceStateImpl) member.getVoiceState();
+        if (voiceState == null)
+            voiceState = new GuildVoiceStateImpl(member);
+        updateGuildVoiceState(voiceState, voiceStateJson, member);
+        return voiceState;
+    }
+
+    private void updateGuildVoiceState(GuildVoiceStateImpl oldVoiceState, DataObject newVoiceStateJson, MemberImpl member)
+    {
+        Guild guild = member.getGuild();
 
-        final long channelId = voiceStateJson.getLong("channel_id");
+        final long channelId = newVoiceStateJson.getLong("channel_id");
         AudioChannel audioChannel = (AudioChannel) guild.getGuildChannelById(channelId);
         if (audioChannel != null)
             ((AudioChannelMixin<?>) audioChannel).getConnectedMembersMap().put(member.getIdLong(), member);
```

Comment:
We should avoid updating the channel connected members cache unless we actually cache voice states, otherwise this cache is incredibly unreliable.